packages/react/src/UnderlineNav/UnderlineNav.figma.tsx

```diff
@@ -28,22 +28,26 @@ figma.connect(
     props: {
       label: figma.textContent('Label'),
       current: figma.enum('state', {
-        selected: '"page"',
+        selected: 'page',
         default: undefined,
-        hover: '"false"',
-        focus: '"false"',
+        hover: 'false',
+        focus: 'false',
       }),
       counter: figma.boolean('counter?', {
         false: undefined,
-        true: figma.children('CounterLabel'),
+        true: figma.string('CounterLabel'),
       }),
       leadingVisual: figma.boolean('leadingIcon?', {
         false: undefined,
         true: figma.children('Icon'),
       }),
     },
     example: ({label, current, counter, leadingVisual}) => (
-      <UnderlineNav.Item aria-current={current} counter={counter} icon={leadingVisual}>
+      <UnderlineNav.Item
+        aria-current={current}
+        counter={counter}
+        icon={leadingVisual ? () => leadingVisual : undefined}
```

Comment:
this does not work I think as the code is not executed. But I'll check.

Reply:
Nope, it does not work and moving it to the figma.boolean section also does not.
I reached out to Figma for this.